File: Shared/DataSheetUtilities.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Module DataSheetUtilities

    Public Function IsPrimaryTypeByGroup(ByVal dr As DataRow) As Boolean

        If (dr(DATASHEET_TRANSITION_TYPE_GROUP_PRIMARY_COLUMN_NAME) Is DBNull.Value) Then
            Return True
        End If

        Return CBool(dr(DATASHEET_TRANSITION_TYPE_GROUP_PRIMARY_COLUMN_NAME))

    End Function

End Module

File: Shared/DataSheetUtilities.cs
// A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System;
using System.Data;
using System.Globalization;

namespace SyncroSim.STSim
{
    internal static class DataSheetUtilities
    {
        public static bool IsPrimaryTypeByGroup(DataRow dr)
        {
            if (dr[Strings.DATASHEET_TRANSITION_TYPE_GROUP_PRIMARY_COLUMN_NAME] == DBNull.Value)
            {
                return true;
            }

            return Convert.ToBoolean(
                dr[Strings.DATASHEET_TRANSITION_TYPE_GROUP_PRIMARY_COLUMN_NAME], 
                CultureInfo.InvariantCulture);
        }
    }
}
